### PR TITLE
(#4071) - more coverage fixes

### PR DESCRIPTION
**Pull request by nolanlawson**
Yet more code coverage fixes, on top of #4077.
Coverage is now up to 93.73%.

Major changes:

1. `COVERAGE=1 npm t` now runs integration, unit, and mapreduce tests
2. `Changes` moved from `utils.js` to its own file, `changesHandler.js`
3. Dead code removed
4. `/* istanbul ignore */` added where it makes sense to ignore code (e.g. `isChromeApp()`)
5. `parseUri.js` simplified
6. General refactoring to move stuff out of `utils.js`.

### CODE REVIEW

**Comment by nolanlawson** [comment]
This PR further reduces the `pouchdb.min.js` size from 49119 to 48721

File: tests/mapreduce/test.mapreduce.js [hunk 1]
```diff
@@ -1169,6 +1169,21 @@ function tests(suiteName, dbName, dbType, viewType) {
...
+                return reject(err);
+              }
+              resolve(res);
+            });
+          }).then(function (res) {
+            res.ok.should.equal(true);
+          });
+        });
+      });
+
```
Comment (nolanlawson) [inline tests/mapreduce/test.mapreduce.js]: ^ this increased the coverage for mapreduce's `callbackify()` function

File: lib/adapters/idb/utils.js [hunk 2]
```diff
@@ -78,7 +79,7 @@ exports.decodeDoc = function (doc) {
   if (!doc) {
     return doc;
   }
-  var idx = utils.lastIndexOf(doc._doc_id_rev, ':');
+  var idx = doc._doc_id_rev.lastIndexOf(':');
```
Comment (nolanlawson) [inline lib/adapters/idb/utils.js]: `String.prototype.lastIndexOf()` is in ES5.

**Comment by nolanlawson** [comment]
rebased

**Comment by daleharvey** [comment]
:+1: awesome work, I am gonna hold off committing since you have a lot in progress and dont want to get you into merge hell

**Comment by nolanlawson** [comment]
There was a bug in Blob creation on old WebKit that our tests caught. Should be fixed now.

**Comment by daleharvey** [comment]
:+1: I have no idea whats up with the .jshintrc though, I cant see any whitespace changes? it should only be a one line diff

**Comment by nolanlawson** [comment]
Yeah I was pretty baffled by that too. I think what happened is that the old file somehow had Windows-style line endings (?) which I removed.

**Pull request merged**